hudi-common/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -56,4 +62,36 @@ public List<String> getRecordKeyFieldNames() {
     throw new UnsupportedOperationException("Bootstrap not supported for key generator. "
         + "Please override this method in your custom key generator.");
   }
+
+  public static String constructRecordKey(String[] recordKeyFields, Function<String, Object> recordKeyValueFunction) {
```

Comment:
nit: should the result of the function be in `String` type so there is no confusion on `object.toString()` in case the `recordKeyValue` is not a `String` object (i.e., let the engine to make sure the value is properly converted, say for byte buffer)?

Reply:
Yes, makes sense to me. Updated the PR